core/src/main/java/com/dtolabs/rundeck/core/storage/StorageTreeFactory.java

```diff
@@ -138,6 +139,10 @@ public StorageTree createTree() throws Exception {
                     ), PathUtil.asPath("/keys")
             );
         }
+        if(null!=fileStorageConverterPluginAdapter && null!=defaultConverters && defaultConverters.contains("FileStorageLayer")){
+            logger.debug("Configuring base converter: FileStorageLayer" );
+            builder=builder.convert(fileStorageConverterPluginAdapter, PathUtil.asPath("/files"));
+        }
```

Comment:
the "defaultConverts" maybe should be refactored completely, allowing the factory resource definition to specify plugin instances/beans to use that way.  It was a small hack to have magic words "KeyStorageLayer", "StorageTimestamperConverter" that added certain converter plugins.  I think we should make a more generic way to do it